woocommerce/woocommerce-ios
# [Order form] Extract logic for ProductStepper from `ProductRowViewModel` into a separate view model

Closes: #11360 


## Why


As discussed in pe5pgL-45I-p2, we'd like to separate the product row in the product selector from the product row in the order form now that the UI has diverged since the product bundles design changes. Right now, the quantity stepper UI is embedded in the product row `ProductRow` under a boolean `canChangeQuantity`, while we also have a separate `SimplifiedProductRow` view with the product & stepper. With the refactoring effort, `ProductRow` should only display the product info without the quantity update UX. As the first step of refactoring, the logic for the quantity update UX is extracted from `ProductRowViewModel` into a separate view model.

## How

In order not to have huge amount of diffs with actual changes for easier code review, I decided to keep the new view model `ProductStepperViewModel` as a property of `ProductRowViewModel`. There are still 2 shared properties between the 2 view models: `quantity` and `removeProductIntent` (a closure that is triggered when the quantity goes to 0) that need to be synced. In the future subtasks of #11357, `ProductRow` and `ProductStepper`(and their view models) should be separated completely and the quantity of `ProductRow` will just be a constant, and `removeProductIntent` won't be part of `ProductRow` because it's only relevant in the order item use case as part of a different & new view model.

Some notes:

- `ProductStepper`'s accessibility label logic was updated so that the entire label string is from the view model `ProductStepperViewModel`, rather than the view model providing just the name of the product stepper
- The stepper-specific test cases were moved to a separate test class

## Testing instructions


- Go to the Orders tab
- Tap on `+` to create an order
- Tap `Add Products` --> the product rows in the product selector should behave the same as before
- Search/select at least one product and tap the bottom CTA to confirm the selection --> the selected product(s) should show quantity as 1
- Expand any product card
- Tap to increment the product quantity --> the quantity label in the price in the product card should also be updated
- After the payments section is synced, tap `Create` to create the order --> the created order should have the recently set quantity for each product
- Tap `Edit` to edit the order --> the quantity for each product should match what was set previously
- Tap to increment the product quantity --> after syncing, the quantity label in the price in the product card should be updated
- Tap to decrease the product quantity to 0 --> the product should be removed
- Tap `Done` --> the updated order should have the recently set quantity for each product 

---

- [x] @jaclync tests the product bundle configuration flow for order creation
- [x] @jaclync tests the product bundle configuration flow for order editing

---
